issue202_test.go

```diff
@@ -0,0 +1,138 @@
+package mergo_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/imdario/mergo"
+)
+
+func TestIssue202(t *testing.T) {
+	tests := []struct {
+		name           string
+		dst, src, want map[string]interface{}
+	}{
+		{
+			name: "slice override string",
+			dst: map[string]interface{}{
+				"x": 456,
+				"y": "foo",
+			},
+			src: map[string]interface{}{
+				"x": "123",
+				"y": []int{1, 2, 3},
+			},
+			want: map[string]interface{}{
+				"x": "123",
+				"y": []int{1, 2, 3},
+			},
+		},
+		{
+			name: "string override slice",
+			dst: map[string]interface{}{
+				"x": 456,
+				"y": []int{1, 2, 3},
+			},
+			src: map[string]interface{}{
+				"x": "123",
+				"y": "foo",
+			},
+			want: map[string]interface{}{
+				"x": "123",
+				"y": "foo",
+			},
+		},
+		{
+			name: "map override string",
+			dst: map[string]interface{}{
+				"x": 456,
+				"y": "foo",
+			},
+			src: map[string]interface{}{
+				"x": "123",
+				"y": map[string]interface{}{
+					"a": true,
+				},
+			},
+			want: map[string]interface{}{
+				"x": "123",
+				"y": map[string]interface{}{
+					"a": true,
+				},
+			},
+		},
```

Comment:
as per example mentioned in the issue, the rest of the cases actually already works without this change.